### ISSUE TITLE
Install package from private Gitlab repo using user/password authentication on Windows

### ISSUE THREAD

**Issue by bartekch** [rstudio/renv#982]
I'm successfully installing/restoring package from Gitlab private repo using Personal Access Token, however I'm wondering whether it is possible to do it using user/password authentication on Windows (this is crucial I believe). Specifically I would like to use Deploy Token, which basically works the same as username + password. I would like to pass both values as environmental variables. I know I could do `renv::install(git::https://GIT_USER:GIT_PASS@gitlab.com/username/package.git)`, but then credentials would be saved in `renv.lock` what is undesirable.

I've tried something like this
```r
Sys.setenv("GIT_USER" = "<username>")
Sys.setenv("GIT_PASS" = "<password>")
renv::install("git::https://gitlab.com/user/package.git")
```
but it's failing with `Error: failed to resolve remote`.  Issue #316 and some digging led me to the conclusion, that it doesn't work because appropriate sections of `renv_scope_git_auth` are not yet implemented for Windows, am I right? I've tested it on Linux and it works.

I've also tried this
```r
remotes::install_git(
  "https://gitlab.com/user/package.git",
  credentials = git2r::cred_env(username = "GIT_USER", password = "GIT_PASS")
)
```
which successfully installed the package, but failed on `renv::restore()`, which wasn't surprising since `renv` has no idea that it should use those credentials.

Is it any way to get around this at this moment? I think it could be useful in some cases, especially in combination with Deploy Tokens.

**Comment by kevinushey** [comment]
> Issue https://github.com/rstudio/renv/issues/316 and some digging led me to the conclusion, that it doesn't work because appropriate sections of renv_scope_git_auth are not yet implemented for Windows, am I right?

I think that's correct. The solution might be simple, though. Right now, `renv` bundles a simple shell script for passing those credentials to git; we'd need a similar batch script for Windows. Basically, the moral equivalent of this:

https://github.com/rstudio/renv/blob/main/inst/resources/scripts-git-askpass.sh

but as a Windows batch script.

**Comment by kevinushey** [comment]
I just pushed a change to `renv` that should try and enable this on Windows:

https://github.com/rstudio/renv/blob/main/inst/resources/scripts-git-askpass.cmd

Are you able to try installing the development version of `renv` and let me know if this works as expected?

**Comment by bartekch** [comment]
Thanks for super quick response!

I'm still getting an error, although different.

```
exiting from: renv_scope_git_auth()
fatal: credential-cache unavailable; no unix socket support
The syntax of the command is incorrect.
error: unable to read askpass response from 'C:/Users/.../renv/resources/scripts-git-askpass.cmd'
fatal: could not read Username for 'https://gitlab.com': terminal prompts disabled
Error: failed to resolve remote 'git::https://gitlab.com/username/package.git' -- error cloning 'package' from 'https://gitlab.com/username/package.git' [status code 128]
```

Unfortunately I don't know how to correct it.

**Comment by kevinushey** [comment]
Hmm... Are you using Git for Windows? I wonder if I can make use of the shell bundled with that here.

**Comment by bartekch** [comment]
Yes, that's what I'm using.

**Comment by kevinushey** [comment]
Can you try once more with https://github.com/rstudio/renv/commit/fb4615b4d5413158439004a98d3b5f32291e7eb2? That seemed to do the trick for me locally on Windows at least.

**Comment by bartekch** [comment]
It did the trick for me either! I've successfully installed and later restored package, worked perfectly.

Although I noticed strange behaviour on the other machine - when restoring, prompt showed up asking for user and password (even in noninteractive session), but when I closed it without providing any data it still managed to cloned the repo and install the package.  I'm not sure what was the reason, but I've updated Git for Windows from 2.33 to the latest version (2.35) and it works good now.

**Comment by kevinushey** [comment]
> It did the trick for me either! I've successfully installed and later restored package, worked perfectly.

Great news; thanks for taking the time to test and confirm!

> Although I noticed strange behaviour on the other machine - when restoring, prompt showed up asking for user and password (even in noninteractive session), but when I closed it without providing any data it still managed to cloned the repo and install the package. I'm not sure what was the reason, but I've updated Git for Windows from 2.33 to the latest version (2.35) and it works good now.

Perhaps the older version of Git for Windows did not respect the `GIT_TERMINAL_PROMPT` environment variable? Either way, if you bump into this again, please let me know and I'll see if I can figure out what's going on.

**Comment by bartekch** [comment]
> Perhaps the older version of Git for Windows did not respect the GIT_TERMINAL_PROMPT environment variable? Either way, if you bump into this again, please let me know and I'll see if I can figure out what's going on.

Possible, I couldn't repeat it so far, but if I do I'll let know. For now  I consider the issue closed. Thanks for your help!

**Comment by bartekch** [comment]
@kevinushey Small update - I bumped into the issue once again, so I checked again and it turns out that the source of the problem was not Git itself, but [Git Credential Manager](https://github.com/GitCredentialManager/git-credential-manager). After I switched it off everything is working as expected. Unfortunately I don't have enough time to pinpoint the exact cause, so I don't know whether this is the issue with this particular credential store or something more general.

I'ne noticed that in `renv_scope_git_auth` there is `GIT_TERMINAL_PROMPT` variable defined, but in `scripts-git-askpass.cmd` `GIT_PROMPT` is used - no idea whether it is related, but maybe?
